### PR TITLE
Add adjustments and color correction to Compatibility renderer

### LINKED ISSUE #66457: OpenGL: Environment adjustments not implemented yet
*Related to https://github.com/godotengine/godot/issues/66455.*

### Godot version

4.0.beta (53d2a9acd)

### System information

Fedora 36, GeForce GTX 1080 (NVIDIA 515.65.01)

### Issue description

Brightness, contrast, saturation sliders and the color correction texture have no effect on the displayed scene.

Both glow and adjustments are enabled on the example below.

### Steps to reproduce

- Switch to the OpenGL driver in the Project Settings.
  - Make sure to start the editor with single-window mode enabled afterwards. This can be done by enabling **Single Window Mode** in the Editor Settings before changing the driver, or by opening the project in the editor directly with the `--single-window` [command line argument](https://docs.godotengine.org/en/latest/tutorials/editor/command_line_tutorial.html).
- Add a WorldEnvironment node, a Camera3D node and a MeshInstance3D in front of the camera.
- Enable glow. Change brightness/contrast/saturation to make a visible difference in the scene.

### Minimal reproduction project

[test_opengl_glow_adjustments_editor.zip](https://github.com/godotengine/godot/files/9648499/test_opengl_glow_adjustments_editor.zip)

### PR DESCRIPTION
**Pull request by clayjohn**
Fixes: https://github.com/godotengine/godot/issues/66457

This is the last of the major features missing from the Compatibility renderer. The remaining features are long term projects (e.g. VoxelGI) or things that won't be supported (e.g. SDFGI).

Using adjustments forces using the post processing pass (which is bad for performance on mobile devices). Originally we wanted to implement this in the scene shader and only use the post processing pass with glow. However, the BCS effects are not suitable for multipass lighting. Brightness, saturation and color correction were broken, but acceptable, but contrast completely breaks to a degree that isn't acceptable. I.e. when using contrast any object with multipass lighting has the reverse of the contrast used.

### CODE REVIEW

File: drivers/gles3/shaders/scene.glsl [hunk 1]
```diff
@@ -1521,6 +1521,7 @@ void main() {
 	if (alpha < alpha_scissor_threshold) {
 		discard;
 	}
+	alpha = 1.0;
```
Comment (clayjohn) [inline drivers/gles3/shaders/scene.glsl:1524]: This fixes a bug that became more obvious as a result of enabling post effects. 

This code matches the behaviour of the RD renderers:
https://github.com/godotengine/godot/blob/11d3768132582d192b8464769f26b493ae822321/servers/rendering/renderer_rd/shaders/forward_clustered/scene_forward_clustered.glsl#L1066-L1068

**Review by BastiaanOlij — APPROVED** [review]
LGTM

**Comment by akien-mga** [comment]
Thanks! Awesome to have the Compatibility renderer basically feature complete for 4.3 beta!

**Comment by Bimbam360** [comment]
Just span up 4.3dev6 and adjustments aren't applying on a newly created scene with a new spatial material and albedo applied:

**Compatibility Behaviour**
compat

At least there's no more errors, but also seemingly does nothing.

For reference:
**Forward Expected Behaviour**
forward

**Comment by naturally-intelligent** [comment]
Thanks for adding this! After glow, this was the last feature missing that brings our project back to full v3 compatibility.

**Pull request merged**